app-android/features/wardrobe/src/main/kotlin/com/closet/features/wardrobe/ClothingFormViewModel.kt

```
@@ -253,224 +293,176 @@ class ClothingFormViewModel @Inject constructor(
+                if (result is DataResult.Success) {
+                    if (isEditMode && originalImagePath != null && originalImagePath != state.imagePath) {
+                        storageRepository.deleteImage(originalImagePath!!)
                     }
-                    else -> { /* No-op */ }
+                    _events.send(ClothingFormEvent.NavigateBack)
+                } else {
+                    _form.update { it.copy(isSaving = false, errorMessage = com.closet.core.ui.R.string.error_database_query) }
                 }
-            } finally {
-                _form.update { it.copy(isSaving = false) }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)